docs/bookstore/page-0.5/pg0.5-env-setup.md

```diff
@@ -0,0 +1,214 @@
+kubectl apply -f config/100-front-end-deployment.yaml
+```
+
+This will create the Deployment and expose it with a Service of type LoadBalancer to receive external traffic:
+
+![Image](images/image11.png)
+
+```shell
+kubectl get pods
+```
+
+You will see that your front end pod is running.
+
+![Image](images/image4.png)
+
+#### **Port Forwarding (Optional under condition)**
+
+![Image](images/image9.png)
+
+You might need to set up port forwarding to access the app from your local machine.
+
+Check if port forwarding is necessary by running:
+
+```shell
+kubectl get services -A
+```
+
+![Image](images/image1.png)
+
+If the `EXTERNAL-IP` for your frontend service is `127.0.0.1`, port forwarding is not needed.
+
+If port forwarding is required, open a new terminal and run:
+
+```shell
+kubectl port-forward svc/bookstore-frontend-svc 3000:3000
+```
+
+![Image](images/image8.png)
+
+Don’t close the terminal when port-forwarding is established.
+
+???+ success "Verify"
+
+    Visit [http://localhost:3000](http://localhost:3000) in your browser. The UI page should appear!
+
+    ![Image](images/image19.png)
+
+### **Task 3: Running the Book Review Service**
+
+![Image](images/image6.png)
+
+The Node.js server is located in the `docs/code-samples/eventing/bookstore-sample-app/start/node-server` folder.
+
+Ensure that port 8080 on your local machine is not being used by another application.
+
+#### **Deploy the Book Review Service: Node.js Server**
+
+You can either [build the image locally](https://docs.docker.com/get-started/02_our_app/) or use our pre-built image. If you are using the pre-built image, you can proceed to the next step.
+
+When ready, run the following command to deploy the Node.js server:
+
+```shell
+kubectl apply -f config/100-deployment.yaml
+```
+
+This command will pull the image and deploy it to your cluster as a Deployment. It will also expose it as a LoadBalancer to receive external traffic.
+
+![Image](images/image15.png)
+
+```shell
+kubectl get pods
+```
+
+You will see that your Node.js server (node-server) pod is running.
+
+![Image](images/image14.png)
+
+#### **Port Forwarding (optional under condition)**
+
+![Image](images/image9.png)
+
+You might need to set up port forwarding to access the app from your local machine.
+
+Check if port forwarding is necessary by running:
+
+```shell
+kubectl get services -A
+```
+
+![Image](images/image7.png)
+
+If the `EXTERNAL-IP` for your Node.js service is `127.0.0.1`, port forwarding is not needed.
+
+If port forwarding is required, open a new terminal and run:
+
+```shell
+kubectl port-forward svc/node-server-svc 8080:80
+```
+
+![Image](images/image17.png)
+
+Don’t close the terminal when port-forwarding is established.
+
+???+ success "Verify"
+
+    Visit [http://localhost:8080](http://localhost:8080) in your browser. The Node.js service should be up and running.
+
+    And in your front end page, you should see the status turns green and say “Connected to node server”.
```

Comment:
I see Hello World
![image](https://github.com/knative/docs/assets/33736985/5f432c09-d80d-440b-acde-ef363fbb5084)

Reply:
Yes! localhost:8080 should say "Hello, world!"

If you go to the front end bookstore page, it should say Connected to node server.

But let me change the wording here to make it more clear @pierDipi

Reply:
Aah, now I get it, but why should I go to the `http://localhost:8080` then?

Reply:
To put it differently, I don't see the value of port-forwarding this other backend server

Reply:
The reason why I do port-forwarding is:

In the front end, we will be sending the cloudevents to the backend.


https://github.com/knative/docs/blob/638cbfbd207be9ce8deee9783db4412a3f2a7670/code-samples/eventing/bookstore-sample-app/solution/frontend/client/components/CommentList.js#L8


https://github.com/knative/docs/blob/638cbfbd207be9ce8deee9783db4412a3f2a7670/code-samples/eventing/bookstore-sample-app/solution/frontend/client/components/CommentForm.js#L167

If we don’t do it, the host name will be an random ip address, and front end will not be able to identify the destination address. But if you have any other approach, please lmk! @pierDipi